packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -499,59 +495,76 @@ export class RunningSummarizer implements IDisposable {
 
         // Wait to generate and send summary
         this.summarizeTimer.start();
+        // Use record type to prevent unexpected value types
         let summaryData: GenerateSummaryData | undefined;
+        let generateTelemetryProps: Record<string, string | number | boolean | undefined> = {};
         try {
             summaryData = await this.internalsProvider.generateSummary({
                 fullTree,
                 refreshLatestAck,
                 summaryLogger: this.logger,
             });
 
-            const {
-                summaryStats,
-                referenceSequenceNumber,
-                ...summaryDataForTelemetry
-            } = summaryData;
-            // Use record type to prevent unexpected value types
-            const telemetryProps: Record<string, string | number | boolean | undefined> = {
-                ...summaryDataForTelemetry,
-                ...summaryStats,
-                refSequenceNumber: referenceSequenceNumber,
-                opsSinceLastAttempt: referenceSequenceNumber - this.heuristics.lastAttempted.refSequenceNumber,
-                opsSinceLastSummary: referenceSequenceNumber - this.heuristics.lastAcked.refSequenceNumber,
+            // Cumulatively add telemetry properties based on how far generateSummary went.
+            const { referenceSequenceNumber: refSequenceNumber } = summaryData;
+            generateTelemetryProps = {
+                refSequenceNumber,
+                opsSinceLastAttempt: refSequenceNumber - this.heuristics.lastAttempted.refSequenceNumber,
+                opsSinceLastSummary: refSequenceNumber - this.heuristics.lastAcked.refSequenceNumber,
             };
-            this.logger.sendTelemetryEvent({
-                eventName: "GenerateSummary",
-                ...telemetryProps,
-            });
+            if (summaryData.stage !== "aborted") {
```

Comment:
This if-else chain is a little confusing since its reversed. Can you please add some comments that might help in understanding this? For instance, it's not obvious that that stage !== "generated" means it's at least "uploaded". Similarly, stage !== "uploaded" means it's "submitted".

Reply:
Also, looks like the stage is used just for logging telemetry. Are you planning to use this elsewhere too? I am thinking if it's worth adding them just for this purpose.

Reply:
We will actually be returning this object out by on-demand summarize call.